### PR TITLE
[plugins/aws][feat] Support more ec2 metrics

### PR DESCRIPTION
**Pull request by meln1k**
# Description

More EC2 metrics are available now

# To-Dos

- [ ] Add test coverage for new or updated functionality
- [ ] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Issues Fixed

- Fixes #XXXX

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

**Review by lloesche — DISMISSED** [review]
Ran it against 40 AWS accounts and it didn't cause any negative impact on runtime. Can also see the extra usage data
```
usage:
  cpu:
    utilization:
      min: 5.637
      avg: 5.914
      max: 8.647
  network:
    in:
      bytes: 20147308704
    out:
      bytes: 8443686926
```

File: resotolib/resotolib/baseresources.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ class BaseResource(ABC):
     _cleaned: bool = False
     _protected: bool = False
     _deferred_connections: List[Dict[str, Any]] = field(factory=list)
-    _resource_usage: Dict[MetricName, Dict[StatName, float]] = field(factory=lambda: defaultdict(dict))
+    _resource_usage: Dict[MetricName, Any] = field(factory=lambda: defaultdict(dict))
```
Comment (aquamatthias) [inline resotolib/resotolib/baseresources.py:159]: Can we make this more explicit?

File: resotocore/resotocore/model/model.py [hunk 2]
```diff
@@ -1476,14 +1469,14 @@ class UsageDatapoint:
...
-        Dictionary of metric names to lists of values. The values are `min`, `avg` and `max`.
+    v: `Dict[str, Any]]`
+        Dictionary of metrics.
 
     """
 
     id: str
     at: int
-    v: Dict[str, UsageMetricValues]
+    v: Dict[str, Any]
```
Comment (aquamatthias) [inline resotocore/resotocore/model/model.py:1479]: Can we make it more explicit?

**Review by aquamatthias — CHANGES_REQUESTED** [review]
Changing the usage metrics structure will break `with_usage`.

**Comment by lloesche** [comment]
I really like this PR ❤️

File: plugins/aws/resoto_plugin_aws/resource/cloudwatch.py [hunk 3]
```diff
@@ -450,3 +450,27 @@ def query_for(
...
+    for query, metric in cloudwatch_result.items():
+        resource = resources_map.get(query.ref_id)
+        if resource is None:
+            continue
+        metric_value = next(iter(metric.metric_values), None)
+        if metric_value is None:
+            continue
+        normalizer = metric_normalizers.get(query.metric_name)
+        if not normalizer:
+            continue
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/cloudwatch.py:471]: metric without normalizer should be thrown away? Is this expected?

**Review by aquamatthias — APPROVED** [review]
thanks

File: plugins/aws/resoto_plugin_aws/resource/cloudwatch.py (hunk 3 above)
Comment (meln1k) [reply plugins/aws/resoto_plugin_aws/resource/cloudwatch.py:471]: that's a neat way to unwrap options here, so yes.

**Pull request merged**